packages/hash/api/src/db/graph/index.ts

```diff
@@ -0,0 +1,536 @@
+import { DataSource } from "apollo-datasource";
+import {
+  Configuration,
+  GraphApi,
+  PropertyType,
+  DataType,
+  LinkType,
+} from "@hashintel/hash-graph-client";
+import HttpAgent, { HttpsAgent } from "agentkeepalive";
+import axios from "axios";
+import { Logger } from "@hashintel/hash-backend-utils/logger";
+import {
+  DbAggregation,
+  DbClient,
+  DbEntity,
+  DbLink,
+  EntityMeta,
+  EntityType,
+  EntityVersion,
+  Graph,
+  VerificationCode,
+} from "../adapter";
+import { SystemType } from "../../types/entityTypes";
+
+const httpAgent = new HttpAgent({
+  maxSockets: 128,
+  maxFreeSockets: 20,
+  timeout: 60 * 1000, // ms
+  freeSocketTimeout: 30 * 1000, // ms
+});
+
+const httpsAgent = new HttpsAgent({
+  maxSockets: 128,
+  maxFreeSockets: 128,
+  timeout: 60000,
+  freeSocketTimeout: 30000,
+});
+
+export class GraphClient extends DataSource implements DbClient {
```

Comment:
It's been a while since I looked at the code but what's the point of us making this new class if all we do is basically call `self.X`? Can we not just use the underlying object?

Reply:
(Although it looks helpful at the moment to see the unimplemented methods)

Reply:
These initial methods will just be proxy calls, but as we get further down the `DbClient`, we'll start seeing places where we'd have to implement filtering logic.

The interface is used by the model classes, which we'd ideally not have to touch after implementing all functionality (there is likely a lot of assumptions about using the Postgres backend, I would guess)